src/SandboxingLevels.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Class SandboxingLevels.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AMP_Comments_Sanitizer;
+use AMP_Form_Sanitizer;
+use AMP_Options_Manager;
+use AMP_Script_Sanitizer;
+use AmpProject\AmpWP\Infrastructure\Conditional;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+
+/**
+ * Experimental service to facilitate flexible AMP.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.2
+ * @internal
+ */
+final class SandboxingLevels implements Service, Registerable, Conditional {
+
+	/**
+	 * Option key for sandboxing level.
+	 *
+	 * @todo Move this to the Options interface once no longer experimental.
+	 * @var string
+	 */
+	const OPTION_SANDBOXING_LEVEL = 'sandboxing_level';
+
+	/**
+	 * Sandboxing levels.
+	 *
+	 * @var int[]
+	 */
+	const SANDBOXING_LEVELS = [ 1, 2, 3 ];
+
+	/**
+	 * Default sandboxing level.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_SANDBOXING_LEVEL = 3;
```

Comment:
In going with a simpler service name of `Sandbox`, I'd suggest using the following constants here instead:
```suggestion
	const OPTION_LEVEL = 'sandboxing_level';

	/**
	 * Sandboxing levels.
	 *
	 * @var int[]
	 */
	const LEVELS = [ 1, 2, 3 ];

	/**
	 * Default sandboxing level.
	 *
	 * @var int
	 */
	const DEFAULT_LEVEL = 3;
```
As an example, `Sandbox::DEFAULT_LEVEL` reads better to me than `SandboxingLevels::DEFAULT_SANDBOXING_LEVEL`.

Or, as another example, this seems to indicate the service itself should be renamed:
`SandboxingLevels::SANDBOXING_LEVELS` => `Sandbox::LEVELS`.

Reply:
Done in a561a2c9fad03a4d27a674d140379f670c5e74d8.